samtranslator/model/sam_resources.py

```diff
@@ -415,6 +412,51 @@ def to_cloudformation(self, **kwargs):  # type: ignore[no-untyped-def] # noqa: P
 
         return resources
 
+    def _make_lambda_role(
+        self,
+        lambda_function: LambdaFunction,
+        intrinsics_resolver: IntrinsicsResolver,
+        execution_role: IAMRole,
+        resources: List[Any],
+        conditions: Dict[str, Any],
+    ) -> None:
+        lambda_role = lambda_function.Role
+
+        if lambda_role is None:
+            resources.append(execution_role)
+            lambda_function.Role = execution_role.get_runtime_attr("arn")
+
+        if is_intrinsic_if(lambda_role):
+            resources.append(execution_role)
+
+            # We need to create and if else condition here
+            role_resolved_value = intrinsics_resolver.resolve_parameter_refs(self.Role)
```

Comment:
To be consistent, let's use the same `lambda_role` instead of `self.Role`. I think they're the same at this point, but it looks more logical to not use `self` (it's not being used anywhere else on this method)